src/core/public/ui_settings/ui_settings_client.ts

```diff
@@ -42,6 +47,9 @@ interface UiSettingsClientParams {
   defaults: Record<string, PublicUiSettingsParams>;
   initialSettings?: UiSettingsState;
   done$: Observable<unknown>;
+  apiForWorkspace: UiSettingsApi;
+  apiForUser: UiSettingsApi;
+  apiForGlobal: UiSettingsApi;
```

Comment:
```suggestion
  uiSettingApis: Record<string, UiSettingsApi>
```
Let's make it a map for extensibility, especially we may introduce scopeService in the future.

Reply:
updated, please have a look. thanks